src/types.rs

```diff
@@ -1,7 +1,7 @@
 /// Represents an input candidate for Coinselection, either as a single UTXO or a group of UTXOs.
 ///
 /// A [`OutputGroup`] can be a single UTXO or a group that should be spent together.
-/// For privacy reasons it might be a good choice to spend a group of UTXOs together.
+/// Grouping UTXOs will downgrade the wallet's privacy heuristics.
```

Comment:
Grouping UTXOs together to be spent actually improves privacy,

Reply:
No. Common input heuristic is one of the main tools of chain analysis. It is expected of a wallet to receive funds in separate UTXOs and group them while spending. When a group of UTXOs are combined to make a spend, it can be reasonably assumed that they are from the same wallet. 
https://medium.com/oxt-research/understanding-bitcoin-privacy-with-oxt-part-2-4-20010e0dab97

Reply:
I guess we have a misunderstanding here.
It is recommended to group UTXOs of the same address(Public key) together.
Because in chain analysis it's a good heuristic to assume that all the `TxIn` in a `Transaction` belong
to a single user. By combining UTXOs of different address together we let the chain know that all of those
belong to a single user wit high probability.

So clubbing UTXOs of the same addresses together is a good practice to enhance privacy.

This method is useful when the user is not using address reuse for every single incoming transactions.

Reply:
@jkciw refer "Evaluating User Privacy in Bitcoin" paper
[Uploading evaluating_user_privacy_in_bitcoin.pdf…]()

Reply:
The idea is if we combine UTXOs from different addresses in the input then it would be obvious to the observer that all those addresses belong to a single user.

Reply:
It is true that combining more than one UTXO belonging to an address is privacy preserving when compared to combining UTXOs belonging to different addresses. Ideally, address reuse should be avoided.